### DISCUSSION TITLE
Are variable fonts supported?

### DISCUSSION THREAD

**Discussion by AlexOkafor** [emilk/egui#1862] [Q&A] [answered]
Hi!
It wasn't clear to what extent otf/ttf was supported:
```rust
let mut fonts = FontDefinitions::default();

// Install my own font (maybe supporting non-latin characters):
fonts.font_data.insert("my_font".to_owned(),
   FontData::from_static(include_bytes!("../../fonts/Ubuntu-Light.ttf"))); // .ttf and .otf supported
```
If I have a font that has bold, semibold, etc built into 1 file does egui support loading those styles from that file?
```
myfont-variable.ttf
```

Or should I be creating separate font families from separate ttf files?
```
myfont-bold.ttf
myfont-semibold.ttf
myfont-thin.ttf
```

Or is there a better configuration to support things like different font weights?

**Comment by AlexOkafor** [comment]
ah I think I found the answer to my own question :)
https://github.com/emilk/egui/blob/0571bf67e29320ea8bf154b3d442de0afacbb5a5/epaint/src/text/fonts.rs#L25
```rust
/// How to select a sized font.
#[derive(Clone, Debug, PartialEq)]
#[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
pub struct FontId {
    /// Height in points.
    pub size: f32,

    /// What font family to use.
    pub family: FontFamily,
    // TODO: weight (bold), italics, …
}
```

**Comment by emilk** [comment, marked as answer]
If there are several font faces in one file, you can use [`FontData::index`](https://docs.rs/epaint/latest/epaint/text/struct.FontData.html#structfield.index) to select the one you want.
